odoo/addons/base/models/ir_actions_report.py

```diff
@@ -564,7 +565,11 @@ def barcode(self, barcode_type, value, **kwargs):
                 if mask_to_apply:
                     mask_to_apply(kwargs['width'], kwargs['height'], barcode)
 
-            return barcode.asString('png')
+            res = barcode.asString(kwargs['format'])
+            if kwargs['format'] == 'svg' and isinstance(res, str):
+                res = res.encode('utf-8')
```

Comment:
To reviewers: the reportlib's `asString` method returns `bytes` for raster images and `str` for svgs
This is here for two reasons: a) to be consistent and b) to be able to use `guess_mimetype` on the result